src/map/Map.tsx

```diff
@@ -30,13 +33,14 @@ export default function({ selectedPath, paths, queryPoints, bbox, mapStyle }: Ma
                 Dispatcher.dispatch(new MapIsLoaded())
             }
         )
-        mapWrapper.fitBounds(bbox)
+        if (prevViewPort)
+            mapWrapper.setViewPort(prevViewPort)
         return () => map?.remove()
     }, [mapStyle])
     useEffect(() => map?.drawPaths(paths, selectedPath), [paths, selectedPath, map])
     useEffect(() => map?.showPathDetails(selectedPath), [selectedPath, map])
     useEffect(() => map?.drawMarkers(queryPoints), [queryPoints, map])
-    useEffect(() => map?.fitBounds(bbox), [bbox, map])
+    useEffect(() => map?.fitBounds(bbox), [bbox])
```

Comment:
This seems to be the most tricky part to me. Keeping the `map` dependency means that after a style change the call to `setMap` will trigger `fitBounds` again. If we calculate a route, scroll somewhere else and then change the style this means the map will center at the calculated route again.

On the other hand if we do not keep the `map` dependency `map` will be undefined here when the page loads initially and the call to `fitBounds` won't happen. Not sure how to fix this tbh.

Reply:
Update: With my last commit it is 'working' now: 2fadb177593d94e0f619888c5707b7d3c422973c, but honestly this seems way to complicated. I now use `useRef` to keep track of the latest bbox value so it is available by the time `onMapReady` is called and we can call `fitBounds(prevBbox.current)`. Is there maybe a better way to construct the map instance before we receive the bbox etc.?

Reply:
Update2: I think I found a better solution here: 933e958d46257890b5c97f375bab797b4a3131cc and now I think the PR is ready to merge.